### PR TITLE
Fixed local_auth crash with API < 24 #24339

### PR DESCRIPTION
**Pull request by diegoveloper**
There are some crashes in these cases:
- When the device has an API lower than 24.  FingerprintManagerCompat was added for those cases.
- When the device has an API greater than 23 but it doesn't have FingerPrint API (weird cases like Samsung J7 neo) , try catch was added for weird cases.
https://github.com/flutter/flutter/issues/24339

### CODE REVIEW

**Comment by diegoveloper** [comment]
cc @mehmetf @zanderso

File: packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java [hunk 1]
```diff
@@ -75,15 +76,31 @@ public void onError(String code, String error) {
...
+          if (fingerprintMgr.isHardwareDetected()) {
+            if (fingerprintMgr.hasEnrolledFingerprints()) {
+              biometrics.add("fingerprint");
+            } else {
+              biometrics.add("undefined");
+            }
+          }
         }
+      } catch (Exception e) {
+        e.printStackTrace();
```
Comment (mehmetf) [inline packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java]: We probably don't want to print stack here and return success. We should:
- Catch the particular exception being thrown
- Report it via result.error()

File: packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java [hunk 2]
```diff
@@ -75,15 +76,31 @@ public void onError(String code, String error) {
...
+          if (fingerprintMgr.isHardwareDetected()) {
+            if (fingerprintMgr.hasEnrolledFingerprints()) {
+              biometrics.add("fingerprint");
+            } else {
+              biometrics.add("undefined");
+            }
+          }
         } else {
-          biometrics.add("undefined");
+          FingerprintManagerCompat fingerprintMgr =
```
Comment (mehmetf) [inline packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java]: If we use the Compat libraries here, do we need to do version check above? Why not just rely on Compat libraries altogether?

**Comment by diegoveloper** [comment]
Yes, I tested on 3 phones.
- essential phone 9.0.
- old samsung galaxy 5.1 without fingerprint.
- Friend's phone ,samsung galaxy j7 neo with android 8.1 without
fingerprint. (I had to add try-catch for this phone , looks like it doesn't
find the fingerprint api on that rom)

**Comment by mehmetf** [comment]
SGTM

**Pull request merged**